native/cocos/platform/empty/modules/SystemWindowManager.cpp

```diff
@@ -0,0 +1,69 @@
+/****************************************************************************
+ Copyright (c) 2021 Xiamen Yaji Software Co., Ltd.
+
+ http://www.cocos.com
+
+ Permission is hereby granted, free of charge, to any person obtaining a copy
+ of this software and associated engine source code (the "Software"), a limited,
+ worldwide, royalty-free, non-assignable, revocable and non-exclusive license
+ to use Cocos Creator solely to develop games on your target platforms. You shall
+ not use Cocos Creator software for developing other software or tools that's
+ used for developing games. You are not granted to publish, distribute,
+ sublicense, and/or sell copies of Cocos Creator.
+
+ The software or tools in this License Agreement are licensed, not sold.
+ Xiamen Yaji Software Co., Ltd. reserves all rights not expressly granted to you.
+
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ THE SOFTWARE.
+ ****************************************************************************/
+
+#include "SystemWindowManager.h"
+#include "platform/BasePlatform.h"
+#include "platform/interfaces/modules/ISystemWindowManager.h"
+#include "platform/empty/modules/SystemWindow.h"
+
+namespace cc {
+
+int SystemWindowManager::init() {
+    return 0;
+}
+
+void SystemWindowManager::processEvent(bool* quit) {
+
+}
+
+void SystemWindowManager::swapWindows() {
+
+}
+
+ISystemWindow *SystemWindowManager::getWindow(uint32_t windowId) const {
+    if (windowId == 0) {
+        return nullptr;
+    }
```

Comment:
Does empty needs to return window handler?

Reply:
Yes, I think it needs.

For example, there are some code in Engine.cpp:

```c++
bool setCanvasCallback(se::Object * /*global*/) {
    se::AutoHandleScope scope;
    se::ScriptEngine *se = se::ScriptEngine::getInstance();
    auto *window = CC_GET_MAIN_SYSTEM_WINDOW();
    auto handler = window->getWindowHandle(); // Get window handle here, if window is nullptr, it will crash.
    auto viewSize = window->getViewSize();
......

    return true;
}
```

This make sure that some logic where needs the ISystemWindow* pointer could go through well .

Reply:
In some mode, there is not a window.
@oahcy please take a look.